### PR TITLE
For #8510. Use SessionConfiguration for createCaptureSession

### PR DESCRIPTION
**Pull request by levimatheri**
@Amejia481 and @csadilek Previous PR got messed up, I do apologize for that. I think my rebase went wrong. Here's a clean PR.
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by Amejia481** [comment]
@levimatheri I added a couple of tests, in some cases the code needed to be reorganized to ease testing.

**Comment by levimatheri** [comment]
> 
> 
> @levimatheri I added a couple of tests, in some cases the code needed to be reorganized to ease testing.

Thank you for helping with that!

File: components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt [hunk 1]
```diff
@@ -495,13 +515,38 @@ class QrFragment : Fragment() {
...
+                listOf(OutputConfiguration(imageSurface), OutputConfiguration(surface)),
+                backgroundExecutor as Executor,
+                stateCallback
+            )
+            camera.createCaptureSession(sessionConfig)
+        } else {
+            @Suppress("DEPRECATION")
+            camera.createCaptureSession(listOf(imageSurface, surface), stateCallback, null)
+        }
+    }
```
Comment (Amejia481) [inline components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt:545]: @levimatheri I create this separate function to test this section by separated without having to provide all the previous dependencies.

Comment (levimatheri) [reply components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt:545]: Aha, that makes sense. Thanks for the link too. Do you have a guide on how to rebase/squash commits properly for PRs?

File: components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt [hunk 2]
```diff
@@ -85,6 +91,66 @@ class QrFragmentTest {
...
+
+        qrFragment.createCaptureSessionCompat(camera, imageSurface, surface, stateCallback)
+
+        @Suppress("DEPRECATION")
+        verify(camera).createCaptureSession(listOf(imageSurface, surface), stateCallback, null)
+    }
+
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.P])
+    fun `WHEN calling createCaptureSessionCompat on a device higher than P THEN use newer api`() {
```
Comment (Amejia481) [inline components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt:133]: @levimatheri I think the issue you were seeing on https://github.com/mozilla-mobile/android-components/pull/9491#discussion_r571672186 was related to the sdk version on which the test was running. Running with an sdk version lower than P the function [will not exist.](https://developer.android.com/reference/android/hardware/camera2/CameraDevice#createCaptureSession(android.hardware.camera2.params.SessionConfiguration))

File: components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt (hunk 1 above)
Comment (Amejia481) [reply components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt:545]: I think [this](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history) could helpful  :)

File: components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt [hunk 3]
```diff
@@ -80,9 +87,70 @@ class QrFragmentTest {
         qrFragment.cameraId = "mockCamera"
         qrFragment.onResume()
         verify(qrFragment, times(2)).startBackgroundThread()
+        verify(qrFragment, times(2)).startExecutorService()
         verify(qrFragment).tryOpenCamera(anyInt(), anyInt(), anyBoolean())
     }
 
+    @Test
+    @Config(sdk = [Build.VERSION_CODES.N])
+    fun `WHEN running a device lower than P THEN startExecutorService should not called executed`() {
```
Comment (csadilek) [inline components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt:96]: nit: should not be called / executed?

**Review by csadilek — APPROVED** [review]
Made another pass, just one nit, but ready from my perspective!

File: components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt (hunk 3 above)
Comment (Amejia481) [reply components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt]: Fixed!

**Review by Amejia481 — APPROVED** [review]
Awesome work @levimatheri, thanks for the patch!

**Comment by levimatheri** [comment]
> 
> 
> Awesome work @levimatheri, thanks for the patch!

Thank you @Amejia481 for the assistance!

**Comment by Amejia481** [comment]
> > Awesome work @levimatheri, thanks for the patch!
> 
> Thank you @Amejia481 for the assistance!

Glad to help :)

**Pull request merged**